be/src/exprs/vectorized/json_functions.cpp

```diff
@@ -150,14 +166,14 @@ bool JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const s
     return ok;
 }
 
-void JsonFunctions::parse_json_paths(const std::string& path_string, std::vector<JsonPath>* parsed_paths) {
+void JsonFunctions::parse_json_paths(const std::string& path_string, std::vector<SimpleJsonPath>* parsed_paths) {
     // split path by ".", and escape quota by "\"
     // eg:
     //    '$.text#abc.xyz'  ->  [$, text#abc, xyz]
     //    '$."text.abc".xyz'  ->  [$, text.abc, xyz]
     //    '$."text.abc"[1].xyz'  ->  [$, text.abc[1], xyz]
```

Comment:
```suggestion
    //    '$."text.abc[1]".xyz'  ->  [$, text.abc[1], xyz]
```

Reply:
Emm, looks like you have some misunderstanding over it.

The `"` in jsonpath is used to quote json field name, like `text.abc`. So `'$."text.abc"[1].xyz'` represents json object `{"text.abc": [{"xyz": 1}]}`. I think the former is the right syntax.

Reply:
@mofeiatwork Will `'$."text.abc"[1].xyz'` be tokennized to `["$", "text.abc", "[1]", "xyz"]`, not `[$, text.abc[1], xyz]` if as it does?

Reply:
`boost::escapted_list_separator` accept three parameters, `Escape character`, `Comma character`, `Quote character`. So above two representation `$."text.abc"[1].xyz` and `$."text.abc[1]".xyz` are same.


```c++
    explicit escaped_list_separator(Char  e = '\\',
                                    Char c = ',',Char  q = '\"')
      : escape_(1,e), c_(1,c), quote_(1,q), last_(false) { }
```